src/app/base.ts

```diff
@@ -138,62 +140,74 @@ export const CommonHeaders = [
     "Upgrade",
     "Via",
     "Warning"
-]
-
-let today = new Date();
-let nextWeek = new Date();
-nextWeek.setDate(today.getDate()+7);
-
-const Tokens = {
-    "{user-mail}": "annew@CIE493742.onmicrosoft.com",
-    "{group-id}": "8f4e3cfd-432d-4b9a-b801-f424aaf08ca1",
-    "{drive-item-id}": "01ZDJCYOZPW7IKQNDL3NHZVRODY2GC2YKW",
-    "{section-id}": "1-fb22b2f1-379f-4da4-bf7b-be5dcca7b99a",
-    "{notebook-id}": "1-fb22b2f1-379f-4da4-bf7b-be5dcca7b99a",
-    "{group-id-with-plan}": "d2b6c7fe-f440-446b-99d8-9ac12e036bf0",
-    "{plan-id}": "RVBp6oJJt0K5f6Lq42zBK2UAA-Rs",
-    "{task-id}": "9jBI6WHDwk60lEWpL4TQ92UAGLhh",
-    "{extension-id}": "com.contoso.roamingSettings",
-    "{host-name}": "cie493742.sharepoint.com",
-    "{server-relative-path}": "sites/Contoso/Operations/Manufacturing",
-    "{group-id-for-teams}": "d2b6c7fe-f440-446b-99d8-9ac12e036bf0",
-    "{channel-id}": "ee3f919c-28e1-4659-86d2-8e37e581335c",
-    "{today}": today.toISOString(),
-    "{next-week}": nextWeek.toISOString(),
-    "AUTHENTICATED_DOMAIN": () => {
-        try {
-            return AppComponent.explorerValues.authentication.user.emailAddress.split("@")[1];
-        } catch(e) {
-            return "example@contoso.com"
+];
+
+export interface Token {
+    placeholder: string
+
+    // base defaults to replace the placeholder with. Not used if any of the below are defined
+    defaultValue?: string
+    defaultValueFn?: Function
+
+    // when the user is not authenticated, use these values for the demo tenant
+    demoTenantValue?: string
+    demoTenantValueFn?: Function
+
+    // when the user is authenticated with MSA or AAD, replace token with these values
+    authenticatedUserValue?: string
+    authenticatedUserValueFn?: Function
+}
+
+/*
+ * Given a token, go through each of the possible replacement scenarios and find which value to
+ * replace the token with.
+ * Order: Authenticated user values, demo tenant replacament values, default replacement values.
+ */
+function getTokenSubstituteValue(token:Token) {
+    let priorityOrder = []; // desc
+
+    if (isAuthenticated()) {
+        priorityOrder.push(token.authenticatedUserValueFn);
+        priorityOrder.push(token.authenticatedUserValue);
+    } else {
+        priorityOrder.push(token.demoTenantValueFn);
+        priorityOrder.push(token.demoTenantValue);
+    }
+
+    priorityOrder.push(token.defaultValueFn);
```

Comment:
np, if the token doesn't have demo tenant or authenticated user values it falls back to the optional default values.